rotkehlchen/chain/arbitrum_one/modules/gmx/decoder.py

```diff
@@ -197,12 +199,49 @@ def decode_position_change(self, context: DecoderContext) -> DecodingOutput:
 
         return DEFAULT_DECODING_OUTPUT
 
+    def _decode_stake(self, context: DecoderContext) -> DecodingOutput:
+        """Decode staking events in the GMX protocol for the GMX asset"""
+        if context.tx_log.topics[0] != STAKE_GMX:
+            return DEFAULT_DECODING_OUTPUT
+
+        account = hex_or_bytes_to_address(context.tx_log.data[0:32])
+        token_addrs = hex_or_bytes_to_address(context.tx_log.data[32:64])
+        amount_raw = hex_or_bytes_to_int(context.tx_log.data[64:96])
+
+        staked_token = self.base.get_or_create_evm_asset(address=token_addrs)
+        amount = asset_normalized_value(amount=amount_raw, asset=staked_token)
+
+        for event in context.decoded_events:
+            if (
+                event.location_label == account and
+                event.event_type == HistoryEventType.SPEND and
+                event.balance.amount == amount
+            ):
+                event.counterparty = CPT_GMX
+                event.event_type = HistoryEventType.STAKING
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                # the contract when staking GMX also creates transfers for (sGMX, sbGMX, sbfGMX)
```

Comment:
why mention this here? What do we need to do for those as developers of the decoder? Is it the elif?

Reply:
added the comment because instead of using GMX as symbol I resolve the asset to use the symbol. The reason is because it catches not only GMX but all the variants that they have until the more staked version of the token